<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Выбор-локации-для-скважины" data-toc-modified-id="Выбор-локации-для-скважины-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Выбор локации для скважины</a></span></li><li><span><a href="#Описание-данных" data-toc-modified-id="Описание-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Описание данных</a></span></li><li><span><a href="#Условия-задачи" data-toc-modified-id="Условия-задачи-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Условия задачи</a></span></li><li><span><a href="#Загрузка-и-подготовка-данных" data-toc-modified-id="Загрузка-и-подготовка-данных-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Загрузка и подготовка данных</a></span><ul class="toc-item"><li><span><a href="#Таблица-region_0" data-toc-modified-id="Таблица-region_0-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Таблица region_0</a></span></li><li><span><a href="#Таблица-region_1" data-toc-modified-id="Таблица-region_1-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Таблица region_1</a></span></li><li><span><a href="#Таблица-region_2" data-toc-modified-id="Таблица-region_2-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Таблица region_2</a></span></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-4.4"><span class="toc-item-num">4.4&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Обучение-и-проверка-модели" data-toc-modified-id="Обучение-и-проверка-модели-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Обучение и проверка модели</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Подготовка-к-расчёту-прибыли" data-toc-modified-id="Подготовка-к-расчёту-прибыли-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Подготовка к расчёту прибыли</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-6.1"><span class="toc-item-num">6.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Расчёт-прибыли-и-рисков" data-toc-modified-id="Расчёт-прибыли-и-рисков-7"><span class="toc-item-num">7&nbsp;&nbsp;</span>Расчёт прибыли и рисков</a></span><ul class="toc-item"><li><span><a href="#Прибыль-региона-0" data-toc-modified-id="Прибыль-региона-0-7.1"><span class="toc-item-num">7.1&nbsp;&nbsp;</span>Прибыль региона 0</a></span></li><li><span><a href="#Прибыль-региона-1" data-toc-modified-id="Прибыль-региона-1-7.2"><span class="toc-item-num">7.2&nbsp;&nbsp;</span>Прибыль региона 1</a></span></li><li><span><a href="#Прибыль-региона-2" data-toc-modified-id="Прибыль-региона-2-7.3"><span class="toc-item-num">7.3&nbsp;&nbsp;</span>Прибыль региона 2</a></span></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-7.4"><span class="toc-item-num">7.4&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-8"><span class="toc-item-num">8&nbsp;&nbsp;</span>Общий вывод</a></span></li><li><span><a href="#Чек-лист-готовности-проекта" data-toc-modified-id="Чек-лист-готовности-проекта-9"><span class="toc-item-num">9&nbsp;&nbsp;</span>Чек-лист готовности проекта</a></span></li></ul></div>

# Выбор локации для скважины

Допустим, вы работаете в добывающей компании «ГлавРосГосНефть». Нужно решить, где бурить новую скважину.

Вам предоставлены пробы нефти в трёх регионах: в каждом 10 000 месторождений, где измерили качество нефти и объём её запасов. Постройте модель машинного обучения, которая поможет определить регион, где добыча принесёт наибольшую прибыль. Проанализируйте возможную прибыль и риски техникой *Bootstrap.*

Шаги для выбора локации:

- В избранном регионе ищут месторождения, для каждого определяют значения признаков;
- Строят модель и оценивают объём запасов;
- Выбирают месторождения с самым высокими оценками значений. Количество месторождений зависит от бюджета компании и стоимости разработки одной скважины;
- Прибыль равна суммарной прибыли отобранных месторождений.

# Описание данных

**Данные геологоразведки трёх регионов находятся в файлах**:
- geo_data_0.csv  
- geo_data_1.csv  
- geo_data_2.csv  

**Данные разбиты на столбцы**:
- id — уникальный идентификатор скважины;  
- f0, f1, f2 — три признака точек (неважно, что они означают, но сами признаки значимы);  
- product — объём запасов в скважине (тыс. баррелей).

# Условия задачи

1) Для обучения модели подходит только линейная регрессия (остальные — недостаточно предсказуемые).  
2) При разведке региона исследуют 500 точек, из которых с помощью машинного обучения выбирают 200 лучших для разработки.  
3) Бюджет на разработку скважин в регионе — 10 млрд рублей.  
4) При нынешних ценах один баррель сырья приносит 450 рублей дохода. Доход с каждой единицы продукта составляет 450 тыс. рублей, поскольку объём указан в тысячах баррелей.  
5) После оценки рисков нужно оставить лишь те регионы, в которых вероятность убытков меньше 2.5%. Среди них выбирают регион с наибольшей средней прибылью.

# Загрузка и подготовка данных

Импортируем необходимые библиотеки

In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
import numpy as np

Загрузим необходимые датасеты

In [2]:
region_0, region_1, region_2 = [pd.read_csv(f'geo_data_{i}.csv') for i in range(3)]

Для просмотра данных о таблицах напишем функицю **df_info**

In [3]:
def df_info(df, name):
    print()
    print(f'Размер таблицы {name}: {df.shape}')
    print()
    print(f'Выведем 10 строк из таблицы {name}')
    display(df.sample(10))
    print()
    print(f'Выведем информацию о таблице {name}')
    print()
    display(df.info())
    print()
    print(f'Посмотрим на статистическую информацию по таблице {name}')
    display(df.describe().T)
    print()
    print(f'Посмотрим есть ли пропущенные значения в таблице {name}')
    display(df.isna().sum())
    print()
    print(f'Количество дубликатов в таблице {name} равно {df.duplicated().sum()} шт')
    print()

## Таблица region_0

Выведем информацию по таблице **region_0**

In [4]:
df_info(region_0, 'region_0')


Размер таблицы region_0: (100000, 5)

Выведем 10 строк из таблицы region_0


id        f0        f1        f2     product
25213  DeJoy  1.214464 -0.568498 -0.163404   65.023997
15970  q8LvI  0.903914  0.490894  4.437696  135.308481
79223  xBmYQ  1.526546 -0.542677  4.171536  111.921701
94578  P3Bwz  0.923451  0.377798  4.333785  130.948776
28968  vM3mT  1.943346  0.190570  2.293551  151.974982
42643  SX4Cw -0.980427 -0.051535  3.603907  123.664919
16864  B7pUR -0.032451  1.061813  3.811593   27.066868
80903  j0D3e  0.972389  0.099969  3.574197  108.811144
7035   04zvT -0.761221  0.610584  0.417089   47.522551
72896  7edZ7  0.075439  1.010225  2.265706   27.555751


Выведем информацию о таблице region_0

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB


None


Посмотрим на статистическую информацию по таблице region_0


count       mean        std        min        25%        50%  \
f0       100000.0   0.500419   0.871832  -1.408605  -0.072580   0.502360   
f1       100000.0   0.250143   0.504433  -0.848218  -0.200881   0.250252   
f2       100000.0   2.502647   3.248248 -12.088328   0.287748   2.515969   
product  100000.0  92.500000  44.288691   0.000000  56.497507  91.849972   

                75%         max  
f0         1.073581    2.362331  
f1         0.700646    1.343769  
f2         4.715088   16.003790  
product  128.564089  185.364347


Посмотрим есть ли пропущенные значения в таблице region_0


id         0
f0         0
f1         0
f2         0
product    0
dtype: int64


Количество дубликатов в таблице region_0 равно 0 шт



## Таблица region_1

Выведем информацию по таблице **region_1**

In [5]:
df_info(region_1, 'region_1')


Размер таблицы region_1: (100000, 5)

Выведем 10 строк из таблицы region_1


id         f0         f1        f2     product
90908  Z0zWl  11.618081  -8.342932  4.000879  107.813044
6587   5hlsu  -7.853967 -11.280069  3.998021  110.992147
18499  HDs9A  11.351424  -1.610640  0.998649   26.953261
54175  e18kG   7.061440  -8.954629  3.005691   80.859783
37702  KoLcL -10.792651  -3.097159  2.001856   57.085625
59090  ahrE9   3.273989  -5.213631  2.002171   53.906522
25070  2tBr6   4.024453  -5.037441  3.001284   80.859783
92228  VnCsP   7.550341  -3.647496  0.997244   26.953261
53343  R22Yf  23.201227  -1.837297  0.997238   26.953261
59079  2lzxt  -6.989271  -1.035253  2.000363   57.085625


Выведем информацию о таблице region_1

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB


None


Посмотрим на статистическую информацию по таблице region_1


count       mean        std        min        25%        50%  \
f0       100000.0   1.141296   8.965932 -31.609576  -6.298551   1.153055   
f1       100000.0  -4.796579   5.119872 -26.358598  -8.267985  -4.813172   
f2       100000.0   2.494541   1.703572  -0.018144   1.000021   2.011479   
product  100000.0  68.825000  45.944423   0.000000  26.953261  57.085625   

                75%         max  
f0         8.621015   29.421755  
f1        -1.332816   18.734063  
f2         3.999904    5.019721  
product  107.813044  137.945408


Посмотрим есть ли пропущенные значения в таблице region_1


id         0
f0         0
f1         0
f2         0
product    0
dtype: int64


Количество дубликатов в таблице region_1 равно 0 шт



## Таблица region_2

Выведем информацию по таблице **region_2**

In [6]:
df_info(region_2, 'region_2')


Размер таблицы region_2: (100000, 5)

Выведем 10 строк из таблицы region_2


id        f0        f1        f2     product
72932  jOEjJ -1.182680 -0.333583 -3.149417   62.674482
23065  2ZUVe -0.300235 -2.605834  3.068233  152.491490
24261  6tMtb -2.075266  1.568280  2.050910  103.987483
30264  DITfO  0.186468 -1.050567  2.952386  111.024628
43799  oT3u3 -1.256399  1.160309  1.328537  133.200473
26917  Nl9Qx -1.103070  0.920230  3.040620   55.561351
62829  hVsf9  0.850860  3.504191  3.831136  175.211766
40769  ylIah -0.983271  1.763975  5.177555   36.988705
21688  jSzPq -2.314402  1.188294  0.636406  134.888513
48761  usNBl -0.915180  0.065581  1.797451   55.376308


Выведем информацию о таблице region_2

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB


None


Посмотрим на статистическую информацию по таблице region_2


count       mean        std        min        25%        50%  \
f0       100000.0   0.002023   1.732045  -8.760004  -1.162288   0.009424   
f1       100000.0  -0.002081   1.730417  -7.084020  -1.174820  -0.009482   
f2       100000.0   2.495128   3.473445 -11.970335   0.130359   2.484236   
product  100000.0  95.000000  44.749921   0.000000  59.450441  94.925613   

                75%         max  
f0         1.158535    7.238262  
f1         1.163678    7.844801  
f2         4.858794   16.739402  
product  130.595027  190.029838


Посмотрим есть ли пропущенные значения в таблице region_2


id         0
f0         0
f1         0
f2         0
product    0
dtype: int64


Количество дубликатов в таблице region_2 равно 0 шт



## Вывод

Как мы видим, во всех таблицах нет пропущенных значений и дубликатов.  
Для дальнейшнего обучения модели машинного обучения необходимо удалить столбец `id` так как он не нужен.

# Обучение и проверка модели

Напишем функцию где разделим наши данны на признаки и целевое значение.  
В нашем случае признаками будут столбцы `f0`, `f1`, `f2`, а целевым значением будет столбец `product` .  Данные будем разбивать на обучающую и валидационную выборки в соотношении 75:25.  
Внутри функции обучим модель **Линейной регресии** и сохраним предсказания и правильные ответы на валидационной выборке.  

In [7]:
def prediction(df):
    
    features = df.drop(['id', 'product'], axis=1)
    target = df['product']
    
    features_train, features_valid, target_train, target_valid = train_test_split(
        features, target, test_size=0.25, random_state=12345)
    
    model = LinearRegression()
    model.fit(features_train, target_train)
    prediction = model.predict(features_valid)
    
    print('Среднее предсказание модели равно {:.2f} тыс. баррелей'.format(prediction.mean()))
    print('Среднеквадратичная ошибка равна {:.2f}'.format(mean_squared_error(target_valid, prediction)**0.5))
    
    return pd.Series(target_valid), pd.Series(prediction)

Сделаем предсказание по таблице **region_0**

In [8]:
target_region_0, pred_region_0 = prediction(region_0)

Среднее предсказание модели равно 92.59 тыс. баррелей
Среднеквадратичная ошибка равна 37.58


Сделаем предсказание по таблице **region_1**

In [9]:
target_region_1, pred_region_1 = prediction(region_1)

Среднее предсказание модели равно 68.73 тыс. баррелей
Среднеквадратичная ошибка равна 0.89


Сделаем предсказание по таблице **region_2**

In [10]:
target_region_2, pred_region_2 = prediction(region_2)

Среднее предсказание модели равно 94.97 тыс. баррелей
Среднеквадратичная ошибка равна 40.03


## Вывод

Из полученных результатов можно сделать вывод, что в среднем наибольшее количество запасов расположено в **region_0** и **region_2**, они равны **92.59** и **94.97** тыс. баррелей, но при этом у них очень большая среднеквадратичная ошибка равная **37.58** и **40.03**, что может говорить о неточности модели.  
В **region_1** среднее значение запасов ниже чем в других регионах и равно **68.73** тыс. баррелей, но при этом среднеквадратичная ошибка равна всего лишь **0.89**, здесь модель явно предсказывает лучше.

# Подготовка к расчёту прибыли

Сохраним исходные данные в переменные

In [11]:
BUDGET = 10**10
NUM_WELLS = 500
GOOD_WELLS = 200
BARREL_INCOME = 450_000

Рассчитаем достаточный объём сырья для безубыточной разработки новой скважины

In [12]:
min_product_level = BUDGET / (BARREL_INCOME * GOOD_WELLS)

print(
    'Достаточный объём сырья для безубыточной разработки новой скважины равен {:.2f} тыс. баррелей'
    .format(min_product_level))

Достаточный объём сырья для безубыточной разработки новой скважины равен 111.11 тыс. баррелей


Теперь сравним полученный объём сырья со средним запасом в каждом регионе

In [13]:
for num, reg in enumerate([pred_region_0, pred_region_1, pred_region_2]):
    if reg.mean() >= min_product_level:
        print(
            f'Средний запас сырья в регионе {num} равен {reg.mean():.2f} тыс. баррелей, такое количество подходит для безубыточной разработки новой скважины'
        )
        print(
            f'Количество скважин с запасом сырья которое больше достаточного объема в регионе {num} равно {(reg>min_product_level).sum()} шт, это составляет {(reg>min_product_level).mean():.2%} от количества скважин в регионе'
        )
        print()
    else:
        print(
            f'Средний запас сырья в регионе {num} равен {reg.mean():.2f} тыс. баррелей, такое количество не подходит для безубыточной разработки новой скважины'
        )
        print(
            f'Количество скважин с запасом сырья которое больше достаточного объема в регионе {num} равно {(reg>min_product_level).sum()} шт, это составляет {(reg>min_product_level).mean():.2%} от количества скважин в регионе'
        )
        print()

Средний запас сырья в регионе 0 равен 92.59 тыс. баррелей, такое количество не подходит для безубыточной разработки новой скважины
Количество скважин с запасом сырья которое больше достаточного объема в регионе 0 равно 5388 шт, это составляет 21.55% от количества скважин в регионе

Средний запас сырья в регионе 1 равен 68.73 тыс. баррелей, такое количество не подходит для безубыточной разработки новой скважины
Количество скважин с запасом сырья которое больше достаточного объема в регионе 1 равно 4594 шт, это составляет 18.38% от количества скважин в регионе

Средний запас сырья в регионе 2 равен 94.97 тыс. баррелей, такое количество не подходит для безубыточной разработки новой скважины
Количество скважин с запасом сырья которое больше достаточного объема в регионе 2 равно 5234 шт, это составляет 20.94% от количества скважин в регионе



## Вывод

Достаточный объём сырья для безубыточной разработки новой скважины равен **111.11 тыс. баррелей**, но среднее значение в каждом регионе ниже этого значения:  
1) В **регионе 0** оно равно **92.59 тыс. баррелей**  
2) В **регионе 1** оно равно **68.73 тыс. баррелей**  
3) В **регионе 2** оно равно **94.97 тыс. баррелей**  
При этом есть скважины с объемом больше требуемого, но их количество составляет около 20% от всех скважин в регионе.

# Расчёт прибыли и рисков 

Напишем функция для расчета прибыли возвращающую значения в млрд рублей

In [14]:
def profit_predict(target, prediction):
    
    sort_prediction = prediction.reset_index(drop=True).sort_values(ascending=False)
    best_target = target.reset_index(drop=True)[sort_prediction.index][:GOOD_WELLS]
    
    mlrd = 10**9
    profit = (best_target.sum()*BARREL_INCOME - BUDGET) / mlrd
    
    return round(profit, 2)

Теперь напишем для функция Bootstrap с 1000 выборок, чтобы найти распределение прибыли.  
Найдем среднюю прибыль, 95%-й доверительный интервал и риск убытков. Убыток — это отрицательная прибыль.

In [15]:
samples= 1000
alpha = 0.025

In [16]:
def bootstrap(target, prediction):
    
    values = []
    state = np.random.RandomState(12345)
    
    for i in range(samples):
        
        target_subsample = target.reset_index(drop=True).sample(
            n=NUM_WELLS, replace=True, random_state=state)
        
        probs_subsample = prediction[target_subsample.index]
        profit = profit_predict(target_subsample, probs_subsample)
        
        values.append(profit)

    values = pd.Series(values)
    lower = values.quantile(alpha)
    confidence_interval = (lower, values.quantile(1-alpha))
    mean = values.mean()
    
    print(f'Средняя выручка: {mean:.2f} млрд. руб')
    print(f'2.5%-квантиль: {lower}')
    print(f'Доверительный интервал: {confidence_interval}')
    print(f'Риск убытка: {(values < 0).mean()} млрд. руб')

## Прибыль региона 0

Применим функция bootstrap для региона 0

In [17]:
bootstrap(target_region_0, pred_region_0)

Средняя выручка: 0.40 млрд. руб
2.5%-квантиль: -0.11
Доверительный интервал: (-0.11, 0.9102499999999998)
Риск убытка: 0.067 млрд. руб


## Прибыль региона 1

Применим функция bootstrap для региона 1

In [18]:
bootstrap(target_region_1, pred_region_1)

Средняя выручка: 0.46 млрд. руб
2.5%-квантиль: 0.03
Доверительный интервал: (0.03, 0.85)
Риск убытка: 0.015 млрд. руб


## Прибыль региона 2

Применим функция bootstrap для региона 2

In [19]:
bootstrap(target_region_2, pred_region_2)

Средняя выручка: 0.40 млрд. руб
2.5%-квантиль: -0.16024999999999998
Доверительный интервал: (-0.16024999999999998, 0.95)
Риск убытка: 0.074 млрд. руб


## Вывод

Мы получили следующую информацию по прибыли и рискам убытка в регионах:
1) В **регионе 0** прибыль равна **0.40 млрд. руб**, с рисками убытка **0.067 млрд. руб**  
2) В **регионе 1** прибыль равна **0.46 млрд. руб**, с рисками убытка **0.015 млрд. руб**    
3) В **регионе 2** прибыль равна **0.40 млрд. руб**, с рисками убытка **0.074 млрд. руб**  

Как мы видим, самую максимальную прибыль можно получить в **регионе 1**, при этом в этом регионе и минимальный риск убытка, поэтому этот регионам необходимо выбрать для разработки скважин

# Общий вывод

В данном проекте нам нужно было выбрать регион, где добыча принесёт наибольшую прибыль.  
Нам были предоставлены пробы нефти в трёх регионах: в каждом 10 000 месторождений, где измерили качество нефти и объём её запасов.  
При изучении данных мы определили, что таблицах нет пропущенных значений и дубликатов.  
Далее мы написали функцию где разделили наши данны на признаки и целевое значение.  
В нашем случае признаками являлись столбцы `f0`, `f1`, `f2`, а целевым значением был столбец `product` .  
Данные разбивали на обучающую и валидационную выборки в соотношении 75:25.  
Внутри функции обучили модель **Линейной регресии** и сохранили предсказания и правильные ответы на валидационной выборке.  

Из полученных результатов можно сделать вывод, что в среднем наибольшее количество запасов расположено в **region_0** и **region_2**, они равны **92.59** и **94.97** тыс. баррелей, но при этом у них очень большая среднеквадратичная ошибка равная **37.58** и **40.03**, что может говорить о неточности модели.  
В **region_1** среднее значение запасов ниже чем в других регионах и равно **68.73** тыс. баррелей, но при этом среднеквадратичная ошибка равна всего лишь **0.89**, здесь модель явно предсказывает лучше.

Достаточный объём сырья для безубыточной разработки новой скважины равен **111.11 тыс. баррелей**, но среднее значение в каждом регионе ниже этого значения:  
1) В **регионе 0** оно равно **92.59 тыс. баррелей**  
2) В **регионе 1** оно равно **68.73 тыс. баррелей**  
3) В **регионе 2** оно равно **94.97 тыс. баррелей**  
При этом есть скважины с объемом больше требуемого, но их количество составляет около 20% от всех скважин в регионе.

Далее мы написали функцию для расчета прибыли и функцию Bootstrap с 1000 выборок, чтобы найти распределение прибыли.
Нашли среднюю прибыль по регионам, 95%-й доверительный интервал и риск убытков. Убыток — это отрицательная прибыль.

В итоге мы получили следующую информацию по прибыли и рискам убытка в регионах:
1) В **регионе 0** прибыль равна **0.40 млрд. руб**, с рисками убытка **0.067 млрд. руб**  
2) В **регионе 1** прибыль равна **0.46 млрд. руб**, с рисками убытка **0.015 млрд. руб**    
3) В **регионе 2** прибыль равна **0.40 млрд. руб**, с рисками убытка **0.074 млрд. руб**  

Как мы видим, самую максимальную прибыль можно получить в **регионе 1**, при этом в этом регионе и минимальный риск убытка, поэтому этот регионам необходимо выбрать для разработки скважин